services/book.services.js

```diff
@@ -111,14 +114,20 @@ async function createBorrowing(book, user_id, lastUpdatedBy, nosAvailable) {
         { where: { id: book.id } },
         { transaction }
       );
+    });
+
+    const allMessages = {
+      userData: userInfo,
+      bookData: book,
+      borrowedBook: borrow,
+    };
 
-      sendBorrowBookEmail(userInfo, book, borrow);
+    await producer.connect();
```

Comment:
You don't have to keep connecting every time you need to send a message from the producer. This connect should only be called once when you are instantiating the producer and not everytime when you are about to send message.